examples/small-study/src/background.js

```diff
@@ -0,0 +1,76 @@
+// Allow outside override for debug and testing using prefs
+const STUDYPREFS = {
+  variationPref: `shield.${addonWidgetId}.testingBranch`,
+  firstrunPref: `shield.${addonWidgetId}.firstrun`,
+};
+
+const STUDYCONFIG = {
+  // the study
+  weightedVariations: {},
+};
+
+// This is a Shield Study template.
+class BaseStudy {
+  constructor(variation) {
+    // also sets activeExperiment key
+    // makes telmemetry possible
+    browser.study.configure(variation, ...STUDYCONFIG);
+  }
+
+  async isEligible() {
+    let permissions = await browser.study.permissions();
+    // could have other reasons to be eligible, such as addons or whatever
+    return permissions.shield;
+  }
+
+  // TODO gets reasons
+  // https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/runtime/onInstalled
+  async installOrDie() {
+    let eligible = await this.isEligible();
+    if (!eligible) {
+      // a full ineligible ending TODO
+      browser.study.endStudy("ineligible" /*, ...*/);
+    }
+  }
+
+  async watchExpire() {
```

Comment:
I claim this should be in the StudyUtils API

Reply:
I claim not, but I am swayable.